src/girouette/tw/border.cljc

```diff
@@ -2,9 +2,19 @@
   (:require [girouette.tw.common :refer [value-unit->css]]))
 
 (def components
-  [{:id :xxx
-    :rules "
-    xxx = <'xxx'> integer
+  [{:id     :border
+    :rules  "
+    border = <'border'> (<'-'> direction)? (<'-'> integer)?
     "
-    :garden (fn [{[val] :component-data}]
-              {:xxx (value-unit->css nil val {})})}])
+    :garden (fn [{:keys [component-data]}]
+              (let [{border-width     :integer :or {border-width "1"}
+                     border-direction :direction} (into {} component-data)
+                    border-prop (case border-direction
+                                  "t" :border-top-width
+                                  "r" :border-right-width
+                                  "b" :border-bottom-width
+                                  "l" :border-left-width
+                                  :border-width)]
+                {border-prop (value-unit->css nil
+                                              [:integer border-width]
+                                              {:number-unit "px"})}))}])
```

Comment:
@raphaelsaunier I usually try to avoid having to construct parsed data like `[:integer border-width]` in the code, to limit the way components are dependent on the common elements of grammar.

The goal is that if one day we change the structure of the grammar for `color`, `lenght`, `lenght-unit` ... we would only need to adjust the utility functions.

Reply:
@raphaelsaunier  I suggest to do something similar to this: 
https://github.com/green-coder/girouette/blob/8947a23f98111e3fffc4acf5a648f26d088558bb/src/girouette/tw/spacing.cljc#L8-L29

You may:
- create an additional rule `border-value` which is optional in the `border` rule,
- then test if it exists using Clojure's `if` and handle its absence as a special case.

Additionally, you may want to extend the grammar to cover more than just an `integer`. How about adding support for `number | length | length-unit` ?

Reply:
Awesome! Thanks for the early feedback. It's super helpful and I'll take that into account for the next rules.

> @raphaelsaunier I usually try to avoid having to construct parsed data like `[:integer border-width]` in the code, to limit the way components are dependent on the common elements of grammar.

Yeah I was wondering how to best achieve this since I didn't see you construct values like this elsewhere in the codebase. What would be the best alternative to generate the `.border` class with no suffixes (which by default generates a `border-width` of `1px`)?

Reply:
I was thinking something like:
```clojure
(if (nil? border-width)
  {border-props "1px"}
  ,,,)
```

Reply:
Oh yes of course, silly me… there's no need to go through the `value-unit->css` function to return this value.